packages/flutter/test/widgets/layout_builder_and_global_keys_test.dart

```diff
@@ -56,4 +57,43 @@ void main() {
       }),
     );
   });
+
+  testWidgets('Moving global key inside a SliverLayoutBuilder', (WidgetTester tester) async {
+    final GlobalKey<StatefulWrapperState> key = GlobalKey<StatefulWrapperState>();
+
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: CustomScrollView(
+          slivers: <Widget>[
+            SliverLayoutBuilder(
+              builder: (BuildContext context, SliverConstraints constraint) {
+                return SliverToBoxAdapter(
+                  child: Wrapper(child: StatefulWrapper(key: key, child: Container(height: 100.0))),
+                );
+              },
+            ),
+          ],
+        ),
+      ),
+    );
+
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: CustomScrollView(
+          slivers: <Widget>[
+            SliverLayoutBuilder(
+              builder: (BuildContext context, SliverConstraints constraint) {
+                key.currentState.trigger();
+                return SliverToBoxAdapter(
+                  child: StatefulWrapper(key: key, child: Container(height: 100.0)),
+                );
+              },
+            ),
+          ],
+        ),
+      ),
+    );
```

Comment:
Are you testing if it does not throw exception? if that is the case, you can do a expect(tester.takeException(), null) at the end